src/github.com/couchbase/sync_gateway/rest/routing.go

```diff
@@ -251,6 +265,22 @@ func wrapRouter(sc *ServerContext, privs handlerPrivs, router *mux.Router) http.
 	})
 }
 
+func matchedOrigin(allowOrigins []string, rqOrigins []string) string {
+	for _, av := range allowOrigins {
+		for _, rv := range rqOrigins {
+			if rv == av {
+				return av
+			}
+		}
+	}
+	for _, av := range allowOrigins {
+		if av == "*" {
+			return "*"
+		}
+	}
+	return ""
+}
+
```

Comment:
If anyone was going to allow "*" in a performance-sensitive setting, I'd suggest we parse allowOrigins when we initially read the CORS property, and set an 'allow all' flag to avoid iterating over the lists every request.  But I'm assuming that's not actually a likely scenario - we wouldn't expect anyone to allow "*" in a production setting?